domains/certificates/Query.API/API/DataSyncSyncer/DataSyncSyncerWorker.cs

```diff
@@ -39,6 +45,20 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             var allContracts = await GetAllContracts(stoppingToken);
             foreach (var contract in allContracts)
             {
+                if (string.IsNullOrEmpty(contract.PublicKey))
+                {
+                    using var scope = scopeFactory.CreateScope();
+                    var keyIssuer = scope.ServiceProvider.GetService<IKeyIssuer>();
+                    var certificateIssuingContractRepository = scope.ServiceProvider.GetService<ICertificateIssuingContractRepository>();
```

Comment:
could this be moved out in a separate service that is run on startup and  then we can delete it afterwards?

Reply:
It was actual the whole code blob as this is per my understanding something that only has to be run once for every existing contract.
I dno if it will fix the scope issue

Reply:
Can't this be fixed with DI like any other service?